File: Personal Practice.java/one-week-of-trouble/bike.py
class Bike:

    def __init__(self):
        self.is_on = False
        self.speed = 0

    def turn_on(self):
        self.is_on = True
        print("Bike is ON")

    def turn_off(self):
        self.is_on = False
        print("Bike is OFF")

    def accelerate(self):

        if self.is_on:

            if self.speed >= 0 and self.speed <= 20:
                self.speed += 1

            elif self.speed >= 21 and self.speed <= 30:
                self.speed += 2

            elif self.speed >= 31 and self.speed <= 40:
                self.speed += 3

            else:
                self.speed += 4

        else:
            print("Bike is OFF")

    def brake(self):

        if self.is_on:

            if self.speed >= 0 and self.speed <= 20:
                self.speed -= 1

            elif self.speed >= 21 and self.speed <= 30:
                self.speed -= 2

            elif self.speed >= 31 and self.speed <= 40:
                self.speed -= 3

            else:
                self.speed -= 4

            if self.speed < 0:
                self.speed = 0

        else:
            print("Bike is OFF")

    def display_speed(self):
        print("Current speed:", self.speed)


bike = Bike()

bike.turn_on()

bike.accelerate()
bike.display_speed()

bike.accelerate()
bike.display_speed()

bike.brake()
bike.display_speed()

bike.turn_off()

File: Personal Practice.java/one-week-of-trouble/Bike.java
public class Bike {

    private boolean isOn;
    private int speed;

    public Bike() {
        isOn = false;
        speed = 0;
    }

    public void turnOn() {
        isOn = true;
        System.out.println("Bike is ON");
    }

    public void turnOff() {
        isOn = false;
        System.out.println("Bike is OFF");
    }

    public void accelerate() {

        if (isOn) {

            if (speed >= 0 && speed <= 20) {
                speed += 1;
            }

            else if (speed >= 21 && speed <= 30) {
                speed += 2;
            }

            else if (speed >= 31 && speed <= 40) {
                speed += 3;
            }

            else {
                speed += 4;
            }
        }

        else {
            System.out.println("Bike is OFF");
        }
    }

    public void brake() {

        if (isOn) {

            if (speed >= 0 && speed <= 20) {
                speed -= 1;
            }

            else if (speed >= 21 && speed <= 30) {
                speed -= 2;
            }

            else if (speed >= 31 && speed <= 40) {
                speed -= 3;
            }

            else {
                speed -= 4;
            }

            if (speed < 0) {
                speed = 0;
            }
        }

        else {
            System.out.println("Bike is OFF");
        }
    }

    public void displaySpeed() {
        System.out.println("Current speed: " + speed);
    }

    public static void main(String[] args) {

        Bike bike = new Bike();

        bike.turnOn();

        bike.accelerate();
        bike.displaySpeed();

        bike.accelerate();
        bike.displaySpeed();

        bike.brake();
        bike.displaySpeed();

        bike.turnOff();
    }
}